tests/subsys/settings/functional/src/settings_basic_test.c

```diff
@@ -0,0 +1,312 @@
+	zassert_true(rc == 2, "_next wrong return value");
+	zassert_not_null(next2, "_next wrong next");
+	zassert_equal_ptr(next2, test2+3, "next points to wrong location");
+
+	/* second separator: return 1, next <> NULL */
+	rc = settings_name_next(next1, &next1);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next1, "_next wrong next");
+	zassert_equal_ptr(next1, test1+5, "next points to wrong location");
+	rc = settings_name_next(next2, &next2);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next2, "_next wrong next");
+	zassert_equal_ptr(next2, test2+5, "next points to wrong location");
+
+	/* third separator: return 1, next <> NULL */
+	rc = settings_name_next(next1, &next1);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next1, "_next wrong next");
+	rc = settings_name_next(next2, &next2);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next2, "_next wrong next");
+
+	/* fourth separator: return 1, next <> NULL */
+	rc = settings_name_next(next1, &next1);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next1, "_next wrong next");
+	rc = settings_name_next(next2, &next2);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_not_null(next2, "_next wrong next");
+
+	/* fifth separator: return 1, next == NULL */
+	rc = settings_name_next(next1, &next1);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_is_null(next1, "_next wrong next");
+	rc = settings_name_next(next2, &next2);
+	zassert_true(rc == 1, "_next wrong return value");
+	zassert_is_null(next2, "_next wrong next");
+
+}
+
+struct stored_data {
+	u8_t val1;
+	u8_t val2;
+	u8_t val3;
+	bool en1;
+	bool en2;
+	bool en3;
+};
+
+struct stored_data data;
+
+int val1_set(const char *key, size_t len, settings_read_cb read_cb,
+	     void *cb_arg)
+{
+	data.val1 = 1;
+	return 0;
+}
+int val1_commit(void)
+{
+	data.en1 = true;
+	return 0;
+}
+static struct settings_handler val1_settings = {
+	.name = "ps",
+	.h_set = val1_set,
+	.h_commit = val1_commit,
+};
+
+int val2_set(const char *key, size_t len, settings_read_cb read_cb,
+	     void *cb_arg)
+{
+	data.val2 = 2;
+	return 0;
+}
+int val2_commit(void)
+{
+	data.en2 = true;
+	return 0;
+}
+static struct settings_handler val2_settings = {
+	.name = "ps/ss/ss",
+	.h_set = val2_set,
+	.h_commit = val2_commit,
+};
+
+int val3_set(const char *key, size_t len, settings_read_cb read_cb,
+	     void *cb_arg)
+{
+	data.val3 = 3;
+	return 0;
+}
+int val3_commit(void)
+{
+	data.en3 = true;
+	return 0;
+}
+static struct settings_handler val3_settings = {
+	.name = "ps/ss",
+	.h_set = val3_set,
+	.h_commit = val3_commit,
+};
+
+/* helper routine to remove a handler from settings */
+int settings_deregister(struct settings_handler *handler)
+{
+	extern sys_slist_t settings_handlers;
+
+	return sys_slist_find_and_remove(&settings_handlers, &handler->node);
+}
+
+static void test_register_and_loading(void)
```

Comment:
nip: looks like this test doesn't ensure storage clearance -  it fails therefore in second run in line 222